pandas/tests/groupby/test_grouping.py

```diff
@@ -814,19 +820,26 @@ def test_groupby_empty(self):
         tm.assert_series_equal(result, expected)
 
         # check group properties
-        assert len(gr.grouper.groupings) == 1
-        tm.assert_numpy_array_equal(
-            gr.grouper.group_info[0], np.array([], dtype=np.dtype(np.intp))
-        )
+        msg = "SeriesGroupBy.grouper is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert len(gr.grouper.groupings) == 1
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_numpy_array_equal(
+                gr.grouper.group_info[0], np.array([], dtype=np.dtype(np.intp))
+            )
 
-        tm.assert_numpy_array_equal(
-            gr.grouper.group_info[1], np.array([], dtype=np.dtype(np.intp))
-        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_numpy_array_equal(
+                gr.grouper.group_info[1], np.array([], dtype=np.dtype(np.intp))
```

Comment:
if for some reason this doesnt change to _grouper, i have a medium-strong preference for doing as little inside the context as possible, so e.g. `gpr = gr.grouper` would go inside the context and the assert_numpy... would go in a separate line